decoder.go

```diff
@@ -25,16 +25,43 @@ type Decoder struct {
 	readTimeout time.Duration
 }
 
-// NewDecoder returns a new Decoder instance that reads events
-// with the given io.Reader.
-func NewDecoder(r io.Reader, readTimeout time.Duration) *Decoder {
+// DecoderOption is a common interface for optional configuration parameters that can be
+// used in creating a Decoder.
+type DecoderOption interface {
+	apply(s *Decoder)
```

Comment:
Unlike some other similar interfaces we've made for variadic options, I didn't allow this to return an `error`. That's because we normally create a `Decoder` within a context (`stream.go` line 274) where "oh wait, your configuration is invalid" is not a situation we could deal with.